bot/exts/evergreen/connect_four.py

```diff
@@ -0,0 +1,377 @@
+        formatted_grid = "\n".join([first_row] + rows)
+        embed = discord.Embed(title="Connect Four Board", description=formatted_grid)
+
+        if self.message:
+            await self.message.edit(embed=embed)
+        else:
+            self.message = await self.channel.send(embed=embed)
+            for emoji in self.unicode_numbers:
+                await self.message.add_reaction(emoji)
+
+    async def start_game(self) -> None:
+        """Begins the game."""
+        self.turn, self.next = self.player1, self.player2
+
+        while True:
+            await self.print_grid()
+            if isinstance(self.turn, AI):
+                coords = self.turn.play()
+            else:
+                coords = await self.player_turn()
+
+            if not coords:
+                return
+
+            if self.check_win(coords, 1 if self.turn == self.player1 else 2):
+                if isinstance(self.turn, AI):
+                    await self.channel.send(f"Game Over! {self.turn.mention} lost against AI")
+                else:
+                    if isinstance(self.next, AI):
+                        await self.channel.send(f"Game Over! {self.turn.mention} won against AI")
+                    else:
+                        await self.channel.send(f"Game Over! {self.turn.mention} won against {self.next.mention}")
+                await self.print_grid()
+                return
+
+            self.turn, self.next = self.next, self.turn
+
+    def predicate(self, reaction: discord.Reaction, user: discord.Member) -> bool:
+        """The predicate to check for the player's reaction."""
+        return (
+            reaction.message.id == self.message.id
+            and user.id == self.turn.id
+            and str(reaction.emoji) in self.unicode_numbers
+        )
+
+    async def player_turn(self) -> Coordinate:
+        """Initiate the player's turn."""
+        message = await self.channel.send(
+            f"{self.turn.mention}, it's your turn! React with a column you want to place your token"
+        )
+        player_num = 1 if self.turn == self.player1 else 2
+        while True:
+            full_column = False
+            try:
+                reaction, user = await self.bot.wait_for("reaction_add", check=self.predicate, timeout=30.0)
+            except asyncio.TimeoutError:
+                await self.channel.send(f"{self.turn.mention}, you took too long. Game over!")
+                return
+            else:
+                await message.delete()
+                await self.message.remove_reaction(reaction, user)
+                column_num = self.unicode_numbers.index(str(reaction.emoji))
+
+                column = [row[column_num] for row in self.grid]
+
+                for row_num, square in reversed(list(enumerate(column))):
+                    if not square:
+                        self.grid[row_num][column_num] = player_num
+                        coords = row_num, column_num
```

Comment:
If you return at this point, you can simplify a large portion of this function. You'd be able to remove all the code that checks for a full column, and all the code that tries to break out of the loop, and the return at the end.